### PR TITLE
FEATURE: Send a private message when a group membership is accepted

### PR DESCRIPTION
**Pull request by nbianca**
Instead of sending the usual notification, it will send a private message, which in turn may send an email.

### CODE REVIEW

File: config/locales/server.en.yml [hunk 1]
```diff
@@ -431,6 +431,10 @@ en:
       trust_level_4: "trust_level_4"
     request_membership_pm:
       title: "Membership Request for @%{group_name}"
+    request_accepted_pm:
+      title: "Membership Request Accepted for @%{group_name}"
+      body: |
+        Your membership request to join @%{group_name} was accepted.
```
Comment (eviltrout) [inline config/locales/server.en.yml:437]:
```suggestion
        Your request to enter @%{group_name} has been accepted and you are now a member.
```

File: spec/requests/groups_controller_spec.rb [hunk 2]
```diff
@@ -1325,6 +1325,23 @@ def expect_type_to_return_right_groups(type, expected_group_ids)
...
+    end
+
+    it "sends a private message when accepted" do
+      group_request = GroupRequest.create!(group: group, user: other_user)
+      expect {
+        put "/groups/#{group.id}/handle_membership_request.json", params: { user_id: other_user.id, accept: true }
+      }.to change { Post.count }.by(1)
+
+      expect(Topic.last.title).to eq(I18n.t('groups.request_accepted_pm.title', group_name: group.name))
+      expect(Post.last.raw).to eq(I18n.t('groups.request_accepted_pm.body', group_name: group.name).strip)
```
Comment (tgxworld) [inline spec/requests/groups_controller_spec.rb:1341]:
```suggestion
      topic = Topic.last
      expect(topic.title).to eq(I18n.t('groups.request_accepted_pm.title', group_name: group.name))
      expect(topic.first_post.raw).to eq(I18n.t('groups.request_accepted_pm.body', group_name: group.name).strip)
```

File: app/models/group.rb [hunk 3]
```diff
@@ -622,16 +622,15 @@ def usernames
...
 
     if notify
       Notification.create!(
         notification_type: Notification.types[:membership_request_accepted],
         user_id: user.id,
-        data: {
-          group_id: id,
-          group_name: name
-        }.to_json
+        data: { group_id: id, group_name: name }.to_json
```
Comment (tgxworld) [inline app/models/group.rb:633]: This change doesn't seem to be required.

File: spec/requests/groups_controller_spec.rb [hunk 4]
```diff
@@ -1325,6 +1325,24 @@ def expect_type_to_return_right_groups(type, expected_group_ids)
...
+    end
+
+    it "sends a private message when accepted" do
+      group_request = GroupRequest.create!(group: group, user: other_user)
+      expect {
+        put "/groups/#{group.id}/handle_membership_request.json", params: { user_id: other_user.id, accept: true }
+      }.to change { Post.count }.by(1)
+
+      topic = Topic.last
+      expect(topic.title).to eq(I18n.t('groups.request_accepted_pm.title', group_name: group.name))
```
Comment (ZogStriP) [inline spec/requests/groups_controller_spec.rb:1341]: I would also test that `topic` is a private message.

**Pull request merged**